src/filters/compress.rs

```diff
@@ -193,14 +193,6 @@ mod tests {
             expected.len(),
             read_context.contents.len()
         );
-        assert_eq!(
```

Comment:
Is it possible to keep the metrics tests - just to ensure we are counting the metrics as we expect to? Or is that not posisble with this refactor?

Reply:
It is possible, it's just more effort. We'd need to move the metrics tests into the integration tests (`tests/`) so that the metrics aren't shared between the tests.

If it's okay, I think I would prefer to add it back (along more tests for metrics in general), in a separate PR(s) since it requires making a lot of visibility changes to expose the metrics publicly so they can be used in the metrics tests. We can make a tracking issue that tracks adding these tests.